src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -77,39 +145,247 @@ public PolicyEntity updatePolicy(@PathParam("domainid") Long domainid, @PathPara
+                query.select(r).where(criteriaBuilder.and(criteriaBuilder.equal(r.get("name"), entity.getName())));
+
+                List<DomainEntity> result = em.createQuery(query).getResultList();
+                if (!result.isEmpty()) {
+                    throw new Exception("Domain name already exists...");
+                    //TODO - to add RETURN 400 error:Sudhir
+                }
+                em.persist(entity);
+                return entity;
+            }
+        }).getId();
+    }
+
+    /**
+     * Updates the Domain for a given domain Id
+     *
+     * @return - updated Domain
+     */
+    @Path("/{domainid}")
+    @PUT
+    public DomainEntity updateDomain(@PathParam("domainid") Long domainid, DomainEntity entity) {
+
+        logger.info("Updating Domain Entity ID...:" + domainid);
+
+        return this.txControl.required(new Callable<DomainEntity>() {
+
+            @Override
+            public DomainEntity call() throws Exception {
+
+                DomainEntity result = em.find(DomainEntity.class, domainid);
+                if (result == null) {
+                    throw new Exception("Domain Entity does not exists...");
+                    //TODO - to add RETURN 404 error:Sudhir
+                }
+                result.setName(entity.getName());
+                em.persist(result);
+                return result;
+            }
+        });
+    }
+
+    /**
+     * Deletes the Domain for a given domain Id
+     *
+     * @return - deleted Domain
+     */
+    @Path("/{domainid}")
+    @DELETE
+    public void deleteDomain(@PathParam("domainid") Long domainid) {
+
+        logger.info("Deleting Domain Entity ID...:" + domainid);
+
+        this.txControl.required(new Callable<Void>() {
+
+            @Override
+            public Void call() throws Exception {
+
+                DomainEntity result = em.find(DomainEntity.class, domainid);
+                if (result == null) {
+                    return null;
+                }
+                em.remove(result);
+                return null;
+            }
+        });
+    }
+
+    /**
+     * Lists the Domain Id's for a given domain
+     *
+     * @return - Domain Id's
+     */
+    @GET
+    public List<String> getDomainIds() {
+
+        logger.info("Listing Domain Ids'");
+
+        return this.txControl.supports(new Callable<List<String>>() {
+
+            @Override
+            public List<String> call() throws Exception {
+
+                CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+                CriteriaQuery<DomainEntity> query = criteriaBuilder.createQuery(DomainEntity.class);
+                Root<DomainEntity> r = query.from(DomainEntity.class);
+                query.select(r);
+                List<DomainEntity> result = em.createQuery(query).getResultList();
+                if (result.isEmpty()) {
+                    throw new Exception("Domain Entity does not exists...");
```

Comment:
if result is empty let it return an empty list.

Reply:
@sudhirappaji I think you missed this comment.

Reply:
OK, yes I missed these comments completely.